### ISSUE TITLE
[Bug] `ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so'`

### ISSUE THREAD

**Issue by vogler** [ClementTsang/bottom#968]
### Checklist

- [X] I've looked through [troubleshooting](https://clementtsang.github.io/bottom/nightly/troubleshooting), [known problems](https://clementtsang.github.io/bottom/nightly/support/official/#known-problems), and  [existing open issues](https://github.com/ClementTsang/bottom/issues?q=is%3Aopen+is%3Aissue) for similar issues.

### Describe the issue

Everything seems to work, but I'm getting the error messages below on a RPi3.

### What operating system and version are you using?

Raspbian GNU/Linux 11 (bullseye)

### What architecture are you using?

arm32

### What version of `bottom` are you running?

0.7.1

### How did you install `bottom`?

snap

### How can we reproduce this?

```console
$ bottom
ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.
ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.
bash: warning: setlocale: LC_ALL: cannot change locale (en_US.UTF-8)
ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.
```

### Additional information

```console
$ uname -a
Linux rpi3 5.15.61-v7+ #1579 SMP Fri Aug 26 11:10:59 BST 2022 armv7l GNU/Linux
$ cat /etc/ld.so.preload
/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so
$ echo $PLATFORM

$ ls /usr/lib/arm-linux-gnueabihf/libarmmem*
/usr/lib/arm-linux-gnueabihf/libarmmem-aarch64.so  /usr/lib/arm-linux-gnueabihf/libarmmem-v7l.so
/usr/lib/arm-linux-gnueabihf/libarmmem-v6l.so      /usr/lib/arm-linux-gnueabihf/libarmmem-v8l.so
$ export PLATFORM=v7l 
$ bottom
ERROR: ld.so: object '/usr/lib/arm-linux-gnueabihf/libarmmem-${PLATFORM}.so' from /etc/ld.so.preload cannot be preloaded (cannot open shared object file): ignored.
...
```

**Comment by ClementTsang** [comment]
I don't have a pi3 to test on and I have no idea about snap, so I don't really know how to help you.

Does the [armv7 binary](https://github.com/ClementTsang/bottom/releases/download/0.7.1/bottom_armv7-unknown-linux-gnueabihf.deb) in the release page run for you at the very least?

**Comment by ClementTsang** [comment]
Also from looking at your errors messages, it seems like the `$PLATFORM` env var isn't set.

**Comment by ClementTsang** [comment]
Also the binary name is `btm`, is that a typo?

**Comment by vogler** [comment]
> Also the binary name is `btm`, is that a typo?

No, for the snap package there's only `bottom`.

> Does the [armv7 binary](https://github.com/ClementTsang/bottom/releases/download/0.7.1/bottom_armv7-unknown-linux-gnueabihf.deb) in the release page run for you at the very least?

Yes, no problem there.

I use `arch=$(dpkg --print-architecture)` in setup scripts to install the right .deb packages for different machines.
This works well for the release filenames as used e.g. in https://github.com/dandavison/delta/releases
However, `armv7` breaks this since `$arch` is `armhf` and `uname -m` is `armv7l`.

**Comment by ClementTsang** [comment]
> No, for the snap package there's only `bottom`.

Yeah, was just asking as the *binary* name is `btm`, not `bottom`, so that just struck me as weird at first.

Other than that... I'm not really sure what to do from here to help you. I don't run the snap distribution, so I can't really make any changes there. I guess I could add `arm-unknown-linux-gnueabihf` builds if that'll help.

**Comment by vogler** [comment]
Yea, the binary is called `bottom` for snap.
Where should I report it to then?

I can also use the .deb for installing, but as I said, the string should be `armhf` as reported by `dpkg`.

**Comment by ClementTsang** [comment]
After thinking about it, probably for the better to switch it back to `.deb` naming conventions. Relevant PR: https://github.com/ClementTsang/bottom/pull/969

That said, it wouldn't fix the snap issue if you prefer that route.

**Comment by vogler** [comment]
> Sorry for the confusion, but is the issue... the file name?

Yes, here's what I meant:
https://github.com/vogler/dotfiles/blob/0313517986a252e612c22c12550b7054e0ebcf7b/install/apt.sh#L34-L39

**Comment by ClementTsang** [comment]
Sounds good. I've already added new files to the 0.7.1 release following standard Debian naming (e.g. `bottom_0.7.1_armhf.deb`) in the meantime while I figure out this build pipeline, if that'll help.

**Comment by ClementTsang** [comment]
Merged #969, if the file naming issue is dealt with I'll close this issue in a bit, as I don't think I can help you regarding the other issue around the snap.

**Comment by vogler** [comment]
Thanks, it's fine, I reported it there and now install from the deb file.

**Comment by vogler** [comment]
To download and install the latest release for your architecture (needs [jq](https://github.com/stedolan/jq)):
```bash
arch=$(dpkg --print-architecture) # amd64, arm64, armhf on RPi (32bit userland)
curl -s https://api.github.com/repos/ClementTsang/bottom/releases/latest | jq ".assets[] | select(.browser_download_url|test(\"$arch\")) | .browser_download_url" -r | wget -q -O /tmp/bottom.deb -i - && sudo dpkg -i /tmp/bottom.deb
```
